src/api/v4/layer/metadata/parser.js

```diff
@@ -0,0 +1,35 @@
+var BucketsMetadata = require('./buckets');
+var CategoriesMetadata = require('./categories');
+var Rule = require('../../../../windshaft-integration/legends/rule.js');
+
+module.exports = {
```

Comment:
minor issue but in order to be consistent:

- private methods are prefixed with underscore
- exported functions are defined in the primary scope (instead inside a module.exports object)
-  module.exports goes at the end of the file and only contains references (not definitions)